tests/testthat/test-verb-joins.R

```diff
@@ -27,6 +27,20 @@ test_that("join with both same and different vars", {
   expect_equal(names(j1), c("x", "y"))
 })
 
+test_that("joining over arbitrary predicates", {
```

Comment:
I think this would be better converted to this style:

```R
test_that("generated sql doesn't change unexpectedly", {
  lf <- lazy_frame(x = 1, y = 2)
  reg <- list(
    inner = inner_join(lf, lf),
    left = left_join(lf, lf),
    right = right_join(lf, lf),
    full = full_join(lf, lf)
  )
  expect_known_output(print(reg), test_path("sql/join.sql"))
})
```